src/daft-scan/src/glob.rs

```diff
@@ -246,56 +247,122 @@ impl ScanOperator for GlobScanOperator {
-        Ok(Box::new(files.map(move |f| {
-            let FileMetadata {
-                filepath: path,
-                size: size_bytes,
-                ..
-            } = f?;
-            Ok(ScanTask::new(
-                vec![DataFileSource::AnonymousDataFile {
-                    path: path.to_string(),
-                    chunk_spec: None,
-                    size_bytes,
-                    metadata: None,
-                    partition_spec: None,
-                    statistics: None,
-                }],
-                file_format_config.clone(),
-                schema.clone(),
-                storage_config.clone(),
-                pushdowns.clone(),
-            )
-            .into())
-        })))
+            // Create one ScanTask per file
+            Ok(Box::new(files.map(move |f| {
+                let FileMetadata {
+                    filepath: path,
+                    size: size_bytes,
+                    ..
+                } = f?;
+                Ok(ScanTask::new(
+                    vec![DataFileSource::AnonymousDataFile {
+                        path: path.to_string(),
+                        chunk_spec: None,
+                        size_bytes,
+                        metadata: None,
+                        partition_spec: None,
+                        statistics: None,
+                    }],
+                    file_format_config.clone(),
+                    schema.clone(),
+                    storage_config.clone(),
+                    pushdowns.clone(),
+                )
+                .into())
+            })))
+        }
+    }
+}
+
+impl GlobScanOperator {
+    async fn split_scan_tasks_by_row_group(
+        &self,
+        files: impl Iterator<Item = DaftResult<FileMetadata>>,
+        io_client: Arc<IOClient>,
+        io_stats: Arc<IOStatsContext>,
+        pushdowns: Pushdowns,
+    ) -> DaftResult<Box<dyn Iterator<Item = DaftResult<ScanTaskRef>> + 'static>> {
+        let handles_iter = files.map(|f| {
+            let io_client = io_client.clone();
+            let io_stats = io_stats.clone();
+            let file_format_config = self.file_format_config.clone();
+            let schema: Arc<daft_core::schema::Schema> = self.schema.clone();
+            let storage_config = self.storage_config.clone();
+            let pushdowns = pushdowns.clone();
+
+            tokio::spawn(async move {
+                let path = f?.filepath;
+
+                let fm =
+                    read_parquet_metadata(&path, io_client.clone(), Some(io_stats.clone())).await?;
+
+                Ok(fm.row_groups.into_iter().enumerate().map(move |(rg, rgm)| {
+                    ScanTask::new(
+                        vec![DataFileSource::AnonymousDataFile {
+                            path: path.to_string(),
+                            chunk_spec: Some(ChunkSpec::Parquet(vec![rg as i64])),
+                            size_bytes: Some(rgm.compressed_size() as u64),
```

Comment:
I interpreted `size_bytes` as representing the file size, which stores the data compressed, so I'm pretty sure `.compressed_size()` is the right method, but if it's the size of the data, then we should use `.total_byte_size()`.